### DISCUSSION TITLE
add implementation of minimum edge clique cover algorithms

### DISCUSSION THREAD

**Discussion by Alex-Markham** [networkx/networkx#6991] [Algorithms]
I would like NetworkX implementations of algorithms for finding a minimum edge clique cover of an undirected graph, for example the exact algorithm proposed by [Gramm et al. 2009](https://dl.acm.org/doi/pdf/10.1145/1412228.1412236) implemented in OCaml and/or the heuristic algorithms of [Conte et al. 2020](https://www.sciencedirect.com/science/article/pii/S089054011930080X) implemented in Java. I have already written ([here](https://gitlab.com/alex-markham/medil/-/blob/db25ac90d2cc28cfc7d84b48da7190e0550aee79/medil/ecc_algorithms.py)) a Python implementation of the former and a Python wrapper that calls the Java code of the latter, but I'd be happy to start by adapting/rewriting my implementation of the exact algorithm into NetworkX. Afterwards (and maybe depending on other people's interest), I'd also be happy to implement Conte et al.'s or some other heuristic algorithm.

**Comment by ghost** [comment]
Those algorithms sound like perfect additions to Networkx
